engineapi/share_manager.go

```diff
@@ -0,0 +1,29 @@
+package engineapi
+
+import (
+	"fmt"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	smclient "github.com/longhorn/longhorn-share-manager/pkg/client"
+)
+
+type ShareManagerClient struct {
+	ip string
+
+	grpcClient *smclient.ShareManagerClient
+}
+
+func NewShareManagerClient(sm *longhorn.ShareManager) (*ShareManagerClient, error) {
+	if sm.Status.State != longhorn.ShareManagerStateRunning {
+		return nil, fmt.Errorf("invalid Share Manager %v, state: %v", sm.Name, sm.Status.State)
+	}
+
+	return &ShareManagerClient{
+		ip:         sm.Status.IP,
```

Comment:
This is control path, so nothing related to storage network case.

Reply:
Addressed. Thank you.